flake.nix

```diff
@@ -12,58 +12,69 @@
+                ${prev.lib.strings.optionalString (solc_version != null) ''
+                  substituteInPlace ./src/kontrol/foundry.py \
+                    --replace "'forge', 'build'," "'forge', 'build', '--no-auto-detect',"
+                ''}
+                substituteInPlace ./pyproject.toml \
+                  --replace ', subdirectory = "kevm-pyk"' ""
+              '';
 
-          overrides = prev.poetry2nix.overrides.withDefaults
-            (finalPython: prevPython: {
-              pyk = prev.pyk-python310;
-              kevm-pyk = prev.kevm-pyk;
-              xdg-base-dirs = prevPython.xdg-base-dirs.overridePythonAttrs
-                (old: {
-                  propagatedBuildInputs = (old.propagatedBuildInputs or [ ])
-                    ++ [ finalPython.poetry ];
+              overrides = prev.poetry2nix.overrides.withDefaults
+                (finalPython: prevPython: {
+                  pyk = prev.pyk-python310;
+                  kevm-pyk = prev.kevm-pyk;
+                  xdg-base-dirs = prevPython.xdg-base-dirs.overridePythonAttrs
+                    (old: {
+                      propagatedBuildInputs = (old.propagatedBuildInputs or [ ])
+                        ++ [ finalPython.poetry ];
+                    });
                 });
-            });
-          groups = [ ];
-          # We remove `"dev"` from `checkGroups`, so that poetry2nix does not try to resolve dev dependencies.
-          checkGroups = [ ];
-
-          postInstall = ''
-            wrapProgram $out/bin/kontrol --prefix PATH : ${
-                prev.lib.makeBinPath [
-                  (solc.mkDefault final final.solc_0_8_13)
-                  final.foundry-bin
-                  prev.which
-                  k-framework.packages.${prev.system}.k
-                ]
-              } --set NIX_LIBS "${nixLibs prev}" --set KEVM_DIST_DIR ${prev.kevm k-framework.packages.${prev.system}.k}
-          '';
-        };
+              groups = [ ];
+              # We remove `"dev"` from `checkGroups`, so that poetry2nix does not try to resolve dev dependencies.
+              checkGroups = [ ];
 
-      };
+              postInstall = ''
+                wrapProgram $out/bin/kontrol --prefix PATH : ${
+                  prev.lib.makeBinPath
+                  ([ prev.which k-framework.packages.${prev.system}.k ]
+                    ++ prev.lib.optionals (solc_version != null) [
+                      final.foundry-bin
+                      (solc.mkDefault final solc_version)
+                    ])
+                } --set NIX_LIBS "${nixLibs prev}" --set KEVM_DIST_DIR ${
+                  prev.kevm k-framework.packages.${prev.system}.k
+                }
+              '';
+            }).overrideAttrs (old: {
+              passthru = old.passthru // (if solc_version == null then {
+                # list all supported solc versions here
+                solc_0_8_13 = kontrol { solc_version = final.solc_0_8_13; };
```

Comment:
if there are any other versions of solc we want to support, we should add them here as:

```nix
solc_0_x_yy = kontrol { solc_version = final.solc_0_x_yy; };
```

Reply:
Can we run multiple versions of solc simultaneously in kup? There are cases in which different versions of solc to compile a solidity source file (depending on the solidity pragma in the source file), thus we need to run multiple versions of solc to compile all the source files in a foundry project. Foundry allows this through automatically installing the required versions.